src/cider/nrepl/middleware/debug.clj

```diff
@@ -207,12 +207,14 @@ this map (identified by a key), and will `dissoc` it afterwards."}
 
 (defn- eval-with-locals
   "`eval` form wrapped in a let of the locals map.
-  If an exception is thrown, it is caught and sent to the client, and
-  this function returns nil."
-  [form locals]
+  If an exception is thrown, it is caught and sent to the client, and this
+  function returns nil. `extras` is a metadata map received from `break'."
+  [form extras]
   (let [ns (ns-name *ns*)]
     (try
-      (binding [*tmp-locals* locals]
+      (binding [*tmp-locals* (:locals extras)
+                ;; evaluate in the instrumentation ns
+                *ns*         (find-ns (symbol (:original-ns extras)))]
```

Comment:
`:original-ns` can never be nil, it's always assigned in `with-initial-debug-bindings`.